### ISSUE TITLE
Should MQTT be an strict requirement of HyperStream?

### ISSUE THREAD

**Issue by perellonieto** [IRC-SPHERE/HyperStream#31]
It is not clear to me what is the role of MQTT in a general framework as HyperStream.
I understand that the original reason is because of our internal use in an Internet of Thinks environment where communication is driven by MQTT, but this requirement may be a barrier for the general user.

Would you think it should be activated optionally or change it to a plugin?

If it is really a required part of HyperStream, we should advertise it in this context then.

**Comment by etonkin** [comment]
Hi,

MQTT should be an optional feature I would say (configurable but not
required). Just another logging method in a pile of possible methods.
It is possibly something that in a different environment I would recommend
configuring via syslog/syslog-ng rather than directly in hyperstream at
all.

To do that meaningfully/safely we'd need to have a good grasp of the
various sorts of information that HyperStream might log.


On 22 February 2018 at 11:02, Miquel Perelló Nieto <notifications@github.com

**Comment by tdiethe** [comment]
Yes agreed - was driven by our requirements. Not sure it’s that easy to pull it out into a plugin though - I did think about doing this at one point. If you succeed I’m all for it.



On 22 February 2018 at 11:39:32, Emma Tonkin (notifications@github.com<mailto:notifications@github.com>) wrote:

Hi,

MQTT should be an optional feature I would say (configurable but not
required). Just another logging method in a pile of possible methods.
It is possibly something that in a different environment I would recommend
configuring via syslog/syslog-ng rather than directly in hyperstream at
all.

To do that meaningfully/safely we'd need to have a good grasp of the
various sorts of information that HyperStream might log.


On 22 February 2018 at 11:02, Miquel Perelló Nieto <notifications@github.com
